packages/user_packages/iotcc_mqtt_dev.py

```diff
@@ -0,0 +1,186 @@
+                        log.error(
+                            '{0} Device relationship with Edge System failed with following error - {1}'.format(
+                                device.name, str(e)))
+                        log.info(
+                            '{0} Device Relationship: Attempt: {1}'.format(device.name, str(relationship_attempts)))
+                        time.sleep(delay_retries)
+                # Use the device name as identifier in the registry to easily refer the device in other packages
+                device_registry_name = socket.gethostname() + "-ChildDev" + str(i)
+                registry.register(device_registry_name, reg_dev)
+
+                # Attempts to set properties for registered device
+                prop_attempts = 0
+                while prop_attempts <= retry_attempts:
+                    try:
+                        self.iotcc.set_properties(reg_dev,
+                                                  {"Country": "USA-G", "State": "California", "City": "Palo Alto",
+                                                   "Location": "VMware HQ", "Building": "Promontory H Lab",
+                                                   "Floor": "First Floor"})
+                        break
+                    except Exception, e:
+                        prop_attempts = prop_attempts + 1
+                        log.error('Exception while setting property for Device {0} - {1}'.format(
+                            (device.name, str(e))))
+                        log.info('Trying setting properties for Device {0}: Attempt - {1}'.format(device.name,
+                                                                                                  str(prop_attempts)))
+                        time.sleep(delay_retries)
+
+                try:
+                    # Registering Metric for Device
+                    self.metrics = []
+                    metric_name = "Simulated Metrics"
+                    metric_simulated_received = Metric(name=metric_name, unit=None, interval=300,
+                                                       aggregation_size=1, sampling_function=dev_metric)
+                    reg_metric_simulated_received = self.iotcc.register(metric_simulated_received)
+                    self.iotcc.create_relationship(reg_dev, reg_metric_simulated_received)
+                    reg_metric_simulated_received.start_collecting()
+                    self.metrics.append(reg_metric_simulated_received)
+
+                except Exception, e:
+                    log.error(
+                        'Exception while loading metric {0} for device {1} - {2}'.format(metric_name, device.name,
+                                                                                         str(e)))
+
+            except Exception:
+                log.info("Device Registration and Metrics loading failed starting the cleanup process")
+                self.clean_up()
```

Comment:
For any Exception at all, do we want to call clean_up() and unregister all the devices completely?

Reply:
As mentioned above I have removed the clean_up call from both the packages.